core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -58,19 +65,86 @@ object ZkUtils extends Logging {
   val IsrChangeNotificationPath = "/isr_change_notification"
   val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
+  
+  
+  def create(zkUrl: String, sessionTimeout: Int, connectionTimeout: Int, loginConfigFile: String): ZkUtils = {
+    val (zkClient, zkConnection) = createZkClientAndConnection(zkUrl, sessionTimeout, connectionTimeout)
+    new ZkUtils(zkClient, zkConnection, loginConfigFile)
+  }
+  
+  def createZkClient(zkUrl: String, sessionTimeout: Int, connectionTimeout: Int): ZkClient = {
+    val zkClient = new ZkClient(zkUrl, sessionTimeout, connectionTimeout, ZKStringSerializer)
+    zkClient
+  }
+
+  def createZkClientAndConnection(zkUrl: String, sessionTimeout: Int, connectionTimeout: Int): (ZkClient, ZkConnection) = {
+    val zkConnection = new ZkConnection(zkUrl, sessionTimeout)
+    val zkClient = new ZkClient(zkConnection, connectionTimeout, ZKStringSerializer)
+    (zkClient, zkConnection)
+  }
+  
+  /*
+   * For tests
+   */
+  def createWithZkClient(zkClient: ZkClient, loginConfigFile: String): ZkUtils = {
+    new ZkUtils(zkClient, null, loginConfigFile)
+  }
+
+  def maybeDeletePath(zkUrl: String, dir: String) {
+    try {
+      val zk = ZkUtils.createZkClient(zkUrl, 30*1000, 30*1000)
+      zk.deleteRecursive(dir)
+      zk.close()
+    } catch {
+      case _: Throwable => // swallow
+    }
+  }
+}
+
+class ZkUtils(val zkClient: ZkClient, 
+              val zkConnection: ZkConnection,
+              val loginConfigFile: String) extends Logging {
   // These are persistent ZK paths that should exist on kafka broker startup.
-  val persistentZkPaths = Seq(ConsumersPath,
-                              BrokerIdsPath,
-                              BrokerTopicsPath,
-                              EntityConfigChangesPath,
-                              ZkUtils.getEntityConfigRootPath(ConfigType.Topic),
-                              ZkUtils.getEntityConfigRootPath(ConfigType.Client),
-                              DeleteTopicsPath,
-                              BrokerSequenceIdPath,
-                              IsrChangeNotificationPath)
+  val persistentZkPaths = Seq(ZkUtils.ConsumersPath,
+                              ZkUtils.BrokerIdsPath,
+                              ZkUtils.BrokerTopicsPath,
+                              ZkUtils.EntityConfigChangesPath,
+                              getEntityConfigRootPath(ConfigType.Topic),
+                              getEntityConfigRootPath(ConfigType.Client),
+                              ZkUtils.DeleteTopicsPath,
+                              ZkUtils.BrokerSequenceIdPath,
+                              ZkUtils.IsrChangeNotificationPath)
+
+  /** true if java.security.auth.login.config is set to some jaas file which has "Client" entry. **/
+
+  val isSecure: Boolean = {
```

Comment:
Would it not be better to receive `isSecure` as a parameter and do the file reading in a factory method?